nltk/tokenize/stanford_segmenter.py

```diff
@@ -43,11 +46,17 @@ class StanfordSegmenter(TokenizerI):
     _JAR = 'stanford-segmenter.jar'
     _SLF4J = 'slf4j-api.jar'
 
-    def __init__(self, path_to_jar=None, path_to_slf4j=None,
-            path_to_sihan_corpora_dict=None,
-            path_to_model=None, path_to_dict=None,
-            encoding='UTF-8', options=None,
-            verbose=False, java_options='-mx2g'):
+    def __init__(self,
+                 path_to_jar=None, path_to_slf4j=None,
+                 java_class='edu.stanford.nlp.ie.crf.CRFClassifier',
+                 path_to_model=None,
+                 path_to_sihan_corpora_dict=None,
+                 sihan_post_processing='true',
+                 keep_whitespaces='false',
+                 path_to_dict=None,
+                 encoding='UTF-8', options=None,
+                 verbose=False, java_options='-mx2g'):
+
```

Comment:
I'm not entire sure of these, as they are for Chinese and I got into this because of Arabic.

From what I can see here:
https://github.com/stanfordnlp/CoreNLP/blob/master/doc/segmenter/README-Chinese.txt
https://github.com/stanfordnlp/CoreNLP/blob/master/src/edu/stanford/nlp/wordseg/ChineseStringUtils.java#L155

the `corpora_dict` is needed for the Chinese model, and `sihan_post_processing` + `keep_whitespaces` have to do with further segmenting numbers & punctuation marks, while `keeping` or collapsing whitespace.